backend/worker.py

```diff
@@ -10,12 +10,45 @@
 from models.post import Post
 from models.invite_user import InviteUser
 from utils import json_response
-from utils import get_all_hierarchy_admin_permissions
 from service_messages import send_message_notification
 from service_messages import send_message_email
 from jinja2 import Environment, FileSystemLoader
 
 
+def should_remove(user, institution_key, current_inst_key):
+    """
+    This method checks whether or not a permission should be removed from the user. 
+    If the user is an administrator of the institution and this institution is 
+    not the one in which the transfer of permissions is being done (Current_inst_key), 
+    it returns false because the permission should not be removed, otherwise it returns true.
+
+    Arguments:
+    user -- User to verify permission
+    institution_key -- Institution key to verify that permissions must be removed
+    current_inst_key -- Key of the institution from which the transfer of permissions is being made.
+    """
+    is_current_inst = institution_key == current_inst_key
+    is_not_admin = (not is_current_inst) and ndb.Key(urlsafe=institution_key) not in user.institutions_admin
+    
+    return is_not_admin or is_current_inst
+
+def filter_permissions_to_remove(user, permissions, institution_key):
+    """
+    This method filters the permissions passed as a parameter and 
+    returns a dictionary of filtered permissions that must be removed 
+    from the user according to the rules applied in the 'should_remove' method.
+
+    Arguments:
+    user -- User to verify if permission must be removed
+    permissions -- Permissions to filter
+    institution_key -- Key of the institution from which the transfer of permissions is being made.
+    """
+    permissions_filtered = {}
+    for permission, insts in permissions.items():
+            inst_keys = [inst for inst in insts if should_remove(user, inst, institution_key)]
```

Comment:
inst -> institution

Use the real name, avoid contractions and misunderstandings.